### PR TITLE
podman: add support for specifying MAC

### PR DESCRIPTION
**Pull request by giuseppe**
rebase of https://github.com/containers/libpod/pull/4058

Signed-off-by: Jakub Filak jakub.filak@sap.com
Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Comment by giuseppe** [comment]
@rhatdan I've tried `vendor-in-container` that I copied from buildah but there is no difference

**Comment by vrothberg** [comment]
> @rhatdan I've tried `vendor-in-container` that I copied from buildah but there is no difference

libpod did not migrate to using go 1.13 for the vendor checks yet. You can run the following command and commit the changes which should unblock this PR:
```bash
$ podman run --privileged --rm --env HOME=/root --env GOPROXY=https://proxy.golang.org -v `pwd`:/src -w /src golang:1.12 make vendor
```

**Comment by giuseppe** [comment]
> $ podman run --privileged --rm --env HOME=/root --env GOPROXY=https://proxy.golang.org -v `pwd`:/src -w /src golang:1.12 make vendor

thanks for the black magic!  It did the trick.  I've updated the patch that adds `vendor-in-container`

File: Makefile [hunk 1]
```diff
@@ -522,6 +522,9 @@ vendor:
 		$(GO) mod vendor && \
 		$(GO) mod verify
 
+vendor-in-container:
+	podman run --privileged --rm --env HOME=/root -v `pwd`:/src -w /src docker.io/library/golang:1.12 make vendor
```
Comment (rhatdan) [inline Makefile]: We want to move everyone to golang 1.13.

@vrothberg Correct?

Comment (giuseppe) [reply Makefile]: we need first to update the CI image, if I use golang:1.13 the vendor check fails

Comment (vrothberg) [reply Makefile]: The CI is currently using 1.12 already which means we can't use 1.13 on our dev machines. The change here is not needed.

**Comment by giuseppe** [comment]
tests are passing

**Comment by giuseppe** [comment]
for reviewers: I've changed this chunk: https://github.com/containers/libpod/pull/4451/files#diff-39e56de7ab505f865271f02541bd0857R840-R855

**Comment by rhatdan** [comment]
LGTM
Really need @mheon to review.
@filak-sap PTAL

File: test/e2e/checkpoint_test.go [hunk 2]
```diff
@@ -334,6 +334,10 @@ var _ = Describe("Podman checkpoint", func() {
 		IPBefore.WaitWithDefaultTimeout()
 		Expect(IPBefore.ExitCode()).To(Equal(0))
 
+		MACBefore := podmanTest.Podman([]string{"inspect", "-l", "--format={{.NetworkSettings.MacAddress}}"})
```
Comment (mheon) [inline test/e2e/checkpoint_test.go]: Can we get a non-checkpoint test? Set `--mac-address ab:cd:ef:gh` and verify that it's set in the container?

**Comment by mheon** [comment]
Code LGTM but I would like another test

**Comment by giuseppe** [comment]
> Code LGTM but I would like another test

test added

**Comment by mheon** [comment]
LGTM

**Comment by rhatdan** [comment]
Awesome job @filak-sap and @giuseppe 
Thanks for getting this done.

**Comment by filak-sap** [comment]
Thank you @giuseppe ! BTW What did I do wrong?

**Pull request merged**